File: c#/leetcode/485-max-consecutive-ones.cs
public class Solution
{
  public int FindMaxConsecutiveOnes(int[] nums)
  {
    int longest = 0;
    int left = 0;

    for (int right = 0; right < nums.Length; right++)
    {
      if (nums[right] == 0)
      {
        left = right + 1;
        continue;
      }
      longest = Math.Max(longest, right - left + 1);
    }
    return longest;
  }
}

File: python/leetcode/485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        count = 0
        values = []

        for i in range(len(nums)):
            if nums[i] == 1:
                count += 1
            if nums[i] == 0:
                values.append(count)
                count = 0

        values.append(count)
        return max(values)


class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:

        left, right = 0, 0
        longest = 0

        for right in range(len(nums)):
            if nums[right] != 1:
                right += 1
                left = right
            else:
                longest = max(longest, (right-left+1))
        return longest
